ssl/statem/statem_clnt.c

```diff
@@ -1130,12 +1133,27 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
             }
         }
 
+        /* Fix session version if it is not reused and we are not in EAP-FAST */
+        if (s->session->tlsext_tick == NULL)
```

Comment:
Hrm. I don't think there are incorrect-version `SSL_SESSION`s to reject in EAP-FAST, actually.

EAP-FAST is basically a messed up version of session tickets, except that you get the ticket and session material from some external mechanism. (That and rather than checking if the server echoed the session ID in ServerHello, you're required to peek ahead for ChangeCipherSpec, which is obnoxious. EAP-FAST doesn't even resolve session resumption until later. See https://github.com/openssl/openssl/blob/master/ssl/statem/statem_clnt.c#L260 But I digress...)

Despite being basically session tickets, OpenSSL's EAP-FAST code isn't structured like this at all. They always make a fresh `SSL_SESSION` and then scribble into the master secret in a callback. See the `tls_session_secret_cb` call. The `ssl_version` isn't filled in from anything, so there's nothing to check. (In fact, if the EAP-FAST consumer even called `SSL_set_session`, the callback would scribble into the master secret which can't possibly work well.)